ddtrace/span.py

```diff
@@ -185,13 +185,20 @@ def to_dict(self):
             'error': self.error,
         }
 
-        # a common mistake is to set the error field to a boolean instead of an
-        # int. let's special case that here, because it's sure to happen in
-        # customer code.
+        # a common mistake is to set some fields without the proper type
+        # and this clashes with the msgpack encoding; to handle that we
+        # convert the field to the right type before encoding
+
+        # error must be an int
         err = d.get('error')
         if err and type(err) == bool:
             d['error'] = 1
 
+        # resource must be a string
+        resource = d.get('resource')
+        if resource and not isinstance(resource, string_type):
+            d['resource'] = stringify(resource)
```

Comment:
Shouldn't we raise an exception instead of blindly converting to a string ? If someone tries to use a value which is not a string, this is a mistake. Forcing a conversion to a string hides the mistake.